### DISCUSSION TITLE
Dependency compilation errors

### DISCUSSION THREAD

**Discussion by pilord12** [csimplestring/delta-go#48] [General]
Hi, I'm trying to play around with using delta-go to read a delta file.  I'm following the example for a local delta table so far, but am getting compilation errors in the s3blob dependency file.  Not really sure why that would be happening, so looking for a little guidance.  Thanks!  Code and errors are below.  Using delta-go v0.0.0-20231105162402-9b93ca02cedf

```
package main

import (
	"fmt"
	delta "github.com/csimplestring/delta-go"
	"log"
	"path/filepath"
)

func main() {
	fmt.Println("hello world")

	path, _ := filepath.Abs("<path-to-delta>")
	path = "file://" + path + "/"
	fmt.Println(path)

	config := delta.Config{
		StoreType: "file",
	}
	table, _ := delta.ForTable(path, config, &delta.SystemClock{})
	snapshot, _ := table.Update()

	version := snapshot.Version()
	log.Println(version)
}

```

```
# gocloud.dev/blob/s3blob
../../go/pkg/mod/gocloud.dev@v0.34.0/blob/s3blob/s3blob.go:411:13: cannot use int32(pageSize) (value of type int32) as *int32 value in struct literal
../../go/pkg/mod/gocloud.dev@v0.34.0/blob/s3blob/s3blob.go:437:15: cannot use obj.Size (variable of type *int64) as int64 value in struct literal
../../go/pkg/mod/gocloud.dev@v0.34.0/blob/s3blob/s3blob.go:592:12: non-boolean condition in if statement
../../go/pkg/mod/gocloud.dev@v0.34.0/blob/s3blob/s3blob.go:721:13: cannot use resp.ContentLength (variable of type *int64) as int64 value in struct literal
../../go/pkg/mod/gocloud.dev@v0.34.0/blob/s3blob/s3blob.go:823:26: cannot use resp.ContentLength (variable of type *int64) as int64 value in argument to getSize
```

**Comment by csimplestring** [comment]
i see your code skip some error check, can you add more error check and show the error messages?

**Reply by pilord12** [reply]
The code is never actually executed, they are not runtime errors, they are compile time.  I'm wondering if it is a dependency mismatch issue.

**Reply by pilord12** [reply]
Yes, I think it was a dependency issue--delta-go uses AWS SDK s3@1.38.1, while my repo was pulling in s3@1.47.2 (which seems to be incompatible with gocloud.dev@v0.34.0).  I forked delta-go and upgraded the gocloud.dev dependency to latest, which seems to fix the issue.  I'll probably make a PR back to delta-go soon.
